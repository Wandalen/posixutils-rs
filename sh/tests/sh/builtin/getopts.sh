set -- -a -b -c
while getopts abc opt; do
    echo $opt
done
echo $OPTIND

OPTIND=1
while getopts abc opt -a -b -c; do
    echo $opt
done
echo $OPTIND

OPTIND=1
while getopts abc opt -a -; do
    echo $opt
done
echo $OPTIND

OPTIND=1
while getopts abc opt -a -- -b -c; do
    echo $opt
done
echo $OPTIND

OPTIND=1
while getopts abc opt -abc; do
    echo $opt
done
echo $OPTIND

OPTIND=1
while getopts a:b:c: opt -a arg1 -b arg2 -c arg3; do
    echo $opt $OPTARG $OPTIND
done
echo $OPTIND

OPTIND=1
while getopts a:b:c: opt -aarg1 -barg2 -carg3; do
    echo $opt $OPTARG $OPTIND
done
echo $OPTIND

OPTIND=1
while getopts :abc opt -d -a -b; do
    echo $opt $OPTARG
done
echo $OPTIND

OPTIND=1
while getopts :a: opt -a; do
    echo $opt $OPTARG
done
echo $OPTIND