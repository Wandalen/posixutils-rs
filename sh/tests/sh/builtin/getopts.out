a
b
c
4
a
b
c
4
a
2
a
3
a
b
c
2
a arg1 3
b arg2 5
c arg3 7
7
a arg1 2
b arg2 3
c arg3 4
4
? d
a
b
4
: a
2
